python/grass/script/tests/grass_script_setup_test.py

```diff
@@ -62,6 +70,32 @@ def test_init_finish_global_functions_with_env(tmp_path):
     assert not os.path.exists(session_file)
 
 
+def init_finish_global_functions_capture_strerr0_partial(tmp_path, queue):
+    gs.set_capture_stderr(True)
+    location = "test"
+    gs.core._create_location_xy(tmp_path, location)  # pylint: disable=protected-access
+    gs.setup.init(tmp_path / location)
+    gs.run_command("g.region", flags="p")
+    runtime_present = bool(os.environ.get("GISBASE"))
+    queue.put((os.environ["GISRC"], runtime_present))
+    gs.setup.finish()
+
+
+def test_init_finish_global_functions_capture_strerr0_partial(tmp_path):
+    """Check that init and finish global functions work with global env using a partial
+    function
+    """
+
+    init_finish = partial(
+        init_finish_global_functions_capture_strerr0_partial, tmp_path
+    )
+    session_file, runtime_present = run_in_subprocess(init_finish)
+    assert session_file, "Expected file name from the subprocess"
+    assert runtime_present, "Runtime (GISBASE) should be present"
+    assert not os.path.exists(session_file), "Session file not deleted"
+
+
+@xfail_mp_spawn
 def test_init_finish_global_functions_capture_strerr0(tmp_path):
```

Comment:
@wenzeslaus what does strerr0 mean? It got changed to it in one of your PRs. If it was intended to be stderr, I would change the name of the new test I made.

Reply:
While this does not strike me as an typo, I don't see nor remember a reason why this should be there, so let's treat that as a typo. If you feel like changing the other ones while doing this overall improvement, please go ahead.

Reply:
Should my new test simply not have strerr0 then?

Reply:
I'll do the renaming on its own, so it's clear that the test are paired, until we have a test only for the different ways of calling the functions, add a duplicate of the tests with our preferred calling method (for cross platform tests), wait a bit, and remove the older tests after some time.

Reply:
Sounds good. Maybe the names are just different variations of the same test, i.e., better name is needed.